stdpopsim/cli.py

```diff
@@ -530,9 +623,50 @@ def run_simulation(args):
             f"Running simulation model {model.id} for {species.id} on "
             f"{contig} with {len(samples)} samples using {engine.id}."
         )
+        if args.dfe is None:
+            dfe = None
+        # Including DFE (selection) in the simulation
+        else:
+            dfe = species.get_dfe(args.dfe)
+            # Including selection in a given interval
+            if args.interval is None:
+                # if not specified, whole contig is under selection
+                contig.add_dfe(
+                    intervals=np.array(
+                        [[0, int(contig.recombination_map.sequence_length)]]
+                    ),
+                    DFE=dfe,
+                )
+                engine = stdpopsim.get_engine(args.engine)  # default to slim
+                logger.info(
+                    f"Running simulation model {model.id} for {species.id} on "
+                    f"{contig} with {len(samples)} samples using {engine.id}."
+                    f"Including selection under the DFE model {dfe.id}."
+                )
+            # Including selection in a specific interval
+            else:
+                contig.add_dfe(
+                    intervals=np.array(
+                        [[0, int(contig.recombination_map.sequence_length)]]
+                    ),
+                    DFE=dfe,
+                )
+                engine = stdpopsim.get_engine(args.engine)
+                logger.info(
+                    f"Running simulation model {model.id} for {species.id} on "
+                    f"{contig} with {len(samples)} samples using {engine.id}."
+                    f"With selection under the DFE model {dfe.id}"
+                    f"in interval {args.interval}."
+                )
```

Comment:
We could simplify things a bit more here, I think:
```suggestion
            dfe = species.get_dfe(args.dfe)
            if args.interval is None:
                # if not specified, whole contig is under selection
                intervals=np.array(
                        [[0, int(contig.recombination_map.sequence_length)]]
                )
            contig.add_dfe(
                    intervals=intervals,
                    DFE=dfe,
            )
            engine = stdpopsim.get_engine(args.engine)
            logger.info(
                    f"Running simulation model {model.id} for {species.id} on "
                    f"{contig} with {len(samples)} samples using {engine.id}."
                    f"With selection under the DFE model {dfe.id}"
                    f"in interval {args.interval}."
            )
```